appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/service/SendKinServiceBase.java

```diff
@@ -6,14 +6,17 @@
 import android.os.IBinder;
 import android.support.annotation.NonNull;
 
+import java.math.BigDecimal;
+
 public abstract class SendKinServiceBase extends Service {
 
     //Developer needs to implements
     //these is called from background thread so can be called async
     public abstract @NonNull
     KinTransferComplete transferKin(@NonNull final String toAddress, final int amount, @NonNull final String memo) throws KinTransferException;
 
-    public abstract int getCurrentBalance() throws BalanceException;
+    public abstract BigDecimal getCurrentBalance() throws BalanceException;
```

Comment:
cast to int
usage of int in the app 
big deciemal has problem with transfering as extra next